lib/drivers/tandemDriver.js

```diff
@@ -1996,6 +2017,99 @@ module.exports = function (config) {
     return records;
   };
 
+  var buildCGMRecords = function(data, records) {
+    var cgmRecords = filterLogEntries([PUMP_LOG_RECORDS.LID_CGM_DATA,PUMP_LOG_RECORDS.LID_CGM_CALIBRATION], data.log_records);
+    for (var b = 0; b < cgmRecords.length; ++b) {
+      var event = cgmRecords[b];
+      var value = event.value;
+      var record = null;
+      var annotation = null;
+
+      if(event.header_id === PUMP_LOG_RECORDS.LID_CGM_CALIBRATION.value
+        && value <=401
+        && value >=39) { // don't include magic numbers
+        record = cfg.builder.makeDeviceEventCalibration();
+
+        record.with_value(value)
+              .with_deviceTime(event.deviceTime)
+              .with_units('mg/dL')      // hard-coded in mg/dL
+              .set('index', event.index)
+              .with_payload({
+                timestamp: event.timestamp,
+                calibration_reading: event.cal_reading,
+                calibration_timestamp: event.cal_timestamp,
+                current_time: event.current_time
+              });
+        cfg.tzoUtil.fillInUTCInfo(record, event.jsDate);
+        record = record.done();
+
+        if(value === 401) {
+           annotation = {
+             code: 'bg/out-of-range',
+             value: 'high',
+             threshold: 400
+           };
+        }
+
+        if(value === 39) {
+           annotation = {
+             code: 'bg/out-of-range',
+             value: 'low',
+             threshold: 40
+           };
+        }
+      }
+      else if (event.header_id === PUMP_LOG_RECORDS.LID_CGM_DATA.value){
+
+        record = cfg.builder.makeCBG();
+
+        if (event.status === CGM_STATUS.CGM_CALIBRATION.value ||
+                 event.status === CGM_STATUS.CGM_LO_CALIBRATION.value ||
+                 event.status === CGM_STATUS.CGM_HI_CALIBRATION.value) {
+          // in response to calibration
+          record.with_payload({
+            calibration_response: true
```

Comment:
From a data model perspective, I'm wondering about this "value in response to calibration" business. Haven't seen discussion of it in the Dexcom specs, in my memory. Have you seen it there?